client/state/sites/reducer.js

```diff
@@ -71,9 +74,12 @@ export function items( state = {}, action ) {
 			// Normalize incoming site(s) to array
 			const sites = action.site ? [ action.site ] : action.sites;
 
+			if ( SITES_UPDATE === action.type && state === null ) {
```

Comment:
Do we need this condition/shortcut? Given how the rest of the function flows, and particularly the condition inside the `reduce` to ignore updates to non-tracked sites, seems like it would just as well without.

Reply:
Hi @aduth, If we don't have this condition, when we have null state and we receive the action SITES_UPDATE, this line of code `const initialNextState = SITES_RECEIVE === action.type || state === null ? {} : state;` is going to set our initial state to empty object (and we will assume sites have been received).
Off course this condition can be changed, to take into account SITES_UPDATE action, but we need a condition to not execute the reduce when initialNextState is null, and the only case when it can stay null is when we receive SITES_UPDATE action so I used that condition to try to make the case explicit.

The other alternative is remove this condition, take SITES_UPDATE  action in consideration when computing initialNextState e.g change .`const initialNextState = SITES_RECEIVE === action.type || state === null ? {} : state;` , and add a condition that skips reduce if initialNextState is null. Let me know if you prefer this version and you think it is more clear, if yes I will make the change.

Reply:
That's fine, I overlooked that we want to preserve the `null` value.